[16.0][MIG] stock_full_location_reservation

mt-software-de opened the pull request:
cc @jbaudoux 

Port of https://github.com/OCA/wms/pull/790

mt-software-de committed:
[ADD] stock_full_location_reservation: Extend reservation to full content of location

mt-software-de committed:
[IMP] stock_full_location_reservation: pre-commit stuff

mt-software-de committed:
[MIG] stock_full_location_reservation: Migration to 16.0

jbaudoux reviewed (approved):
Thanks
Failing test does not seem related to this PR

jbaudoux commented:
cc @rousseldenis

rousseldenis commented:
@mt-software-de Could you rebase ?

rousseldenis commented:
@mt-software-de Do you plan to migrate shopfloor module too ?

rousseldenis commented:
> @mt-software-de Do you plan to migrate shopfloor module too ?

I'll do it

mt-software-de commented:
> @mt-software-de Do you plan to migrate shopfloor module too ?

yes

mt-software-de commented:
> @mt-software-de Could you rebase ?

done

rousseldenis committed:
[IMP] stock_full_location_reservation: Be able to merge the new movement for the whole quantity

rousseldenis committed:
[IMP] stock_full_location_reservation: Prevent first confirmation to merge

As the _action_confirm() will do merges and as new moves are confirmed now, this can lead
to current move unlink. So, don't merge on new move confirmation, unreserve current move,
then process to merge.

rousseldenis committed:
[IMP] stock_full_location_reservation: Improve README

rousseldenis committed:
[IMP] stock_full_location_reservation: Move line should have been deleted

jbaudoux commented:
As commented on v14.0 PR: https://github.com/OCA/wms/pull/790#pullrequestreview-1778040443
Can we make this more generic and refactor a bit into a stock_full_source_reservation.
With the location content transfer, we want to trigger this full location reservation.
However, on a zone picking or single product transfer, we want to extend the reservation to what is being moved. If you move a product or a lot, you want to move the full product/lot at source location. If you move a package, you want to move the full package from source location. If you have multiple products or packages on the source location, you don't want to book them all.

rousseldenis commented:
@jbaudoux Can we move forward and maybe refactor in a further PR ?

jbaudoux commented:
> @jbaudoux Can we move forward and maybe refactor in a further PR ?

@rousseldenis Sure. I had approved, you did not yet ;)

mt-software-de commented:
> > @jbaudoux Can we move forward and maybe refactor in a further PR ?
> 
> @rousseldenis Sure. I had approved, you did not yet ;)

Could you please also merge v14 #790

rousseldenis commented:
/ocabot merge nobump
